### PR TITLE
(PA-3264) Replace facter 3 with facter 4

### PR DESCRIPTION
**Pull request by florindragos**
This PR removes cfacter and replaces it with facter 4

Tested on ubuntu 18.04:
`root@unclaimed-comet:~# /opt/puppetlabs/puppet/bin/puppet facts | grep facterversion
    "facterversion": "4.0.26",`

### CODE REVIEW

File: acceptance/tests/ensure_facter_command_can_be_executed.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+test_name 'Ensure facter command can be executed' do
+
+  confine :to, :platform => /no-platform-yet/
```
Comment (gimmyxd) [inline acceptance/tests/ensure_facter_command_can_be_executed.rb:3]: Should this be enabled now? as facter command should be available on all boxes

**Comment by gimmyxd** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos7-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1804-64a,windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a

**Comment by gimmyxd** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos7-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1804-64a,windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

**Comment by gimmyxd** [comment]
jenkins please test this on redhat7-POWER,aix61-POWERf,sles12-POWER,ubuntu1604-POWER

**Comment by florindragos** [comment]
jenkins please test this on redhat7-POWER,aix61-POWERf,sles12-POWER,ubuntu1604-POWER

**Comment by florindragos** [comment]
jenkins please test this on redhat7-POWER,aix61-POWERf,sles12-POWER,ubuntu1604-POWER

**Comment by florindragos** [comment]
jenkins please test this on redhat7-POWER,aix61-POWERf,sles12-POWER,ubuntu1604-POWER

**Comment by florindragos** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos7-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1804-64a,windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

**Comment by florindragos** [comment]
jenkins please test this on windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

**Comment by florindragos** [comment]
jenkins please test this on windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

**Comment by florindragos** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos7-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1804-64a,windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

**Comment by florindragos** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos7-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1804-64a,windows10ent-64a,windows2012r2-64a,windows2016-64a,windows2019-64a with facter,puppet

File: acceptance/tests/ensure_facter_command_can_be_executed.rb (hunk 1 above)
Comment (gimmyxd) [reply acceptance/tests/ensure_facter_command_can_be_executed.rb:3]: @florindragos can this be enabled now?

**Pull request merged**